ui/migrations/0036_video_duration.py

```diff
@@ -0,0 +1,62 @@
+# Generated by Django 4.2.21 on 2025-05-23 01:10
+
+from ast import literal_eval
+from django.db import migrations, models, connection
+
+
+def migrate_video_duration(apps, schema_editor):
+    """
+    Migrate video duration from the old encode job if data exists.
+    """
+
+    if (
+        "dj_elastictranscoder_encodejob"
+        in schema_editor.connection.introspection.table_names()
+    ):
+        Video = apps.get_model("ui", "Video")
+        db_alias = schema_editor.connection.alias
+
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                SELECT e1.object_id, e1.message 
+                FROM dj_elastictranscoder_encodejob e1
+                INNER JOIN (
+                    SELECT object_id, MAX(created_at) as max_created_at
+                    FROM dj_elastictranscoder_encodejob
+                    WHERE state = 4
+                    GROUP BY object_id
+                ) e2 ON e1.object_id = e2.object_id AND e1.created_at = e2.max_created_at
+                WHERE e1.state = 4
+                """
+            )
+            rows = cursor.fetchall()
+
+        for row in rows:
+            video_id, message = row
+            if not (message := literal_eval(message)):
+                continue
+
+            duration = isinstance(message, dict) and message.get("Output", {}).get(
```

Comment:
Since this is a new field on the `Video` model, shouldn't this migration also assign a duration value to videos encoded with the new mediaconvert workflow?

Currently this only seems to apply to videos that were encoded using the old encoding workflow where there is an `{"Output": {"Duration": x}}` field, but the videos generated with the new mediaconvert workflow seem to have `{"outputGroupDetails": [{"outputDetails": [{"durationInMs": x}]]}` instead.